File: c_lang/src/TwoSum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int i = 0;
        int size = nums.size();
        while(i < size){
            int numberChecking = target- nums[i];
            int j = size-1;
            while(j>0){
                if(j==i){
                    j--;
                    continue;
                }
                if(nums[j]==numberChecking){
                    vector<int> result{i,j};
                    return result;
                }
                j--;
            }
            i++;
        }
        vector<int> result{};
        return result;
    }
};

File: rust_lang/src/solutions/twosum.rs
use std::collections::HashMap;

#[derive(Debug)]
struct Solution {}

impl Solution {
    fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut hash: HashMap<i32, i32> = HashMap::new();
        for (i, value) in nums.iter().enumerate() {
            let find_value = target - value;
            if hash.contains_key(&find_value) {
                return vec![hash[&find_value], (i as i32)];
            }
            hash.insert(value.clone(), i as i32);
        }
        return vec![];
    }
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn it_works() {
        let result = Solution::two_sum(vec![1, 2, 3, 4], 7);
        println!("{:?}", result);
    }
}
